packages/fleet/lib/fleet.ts

```diff
@@ -155,4 +155,26 @@ export class Fleet {
     public async idleNode({ nodeId }: { nodeId: number }): Promise<Result<Node>> {
         return nodes.idle(this.dbClient.db, { nodeId });
     }
+
+    public async overrideNodeConfig(override: Omit<NodeConfigOverride, 'id' | 'createdAt' | 'updatedAt'>): Promise<Result<NodeConfigOverride>> {
+        const defaultConfig = this.nodeProvider.defaultNodeConfig;
+        const search = await nodeConfigOverrides.search(this.dbClient.db, { routingIds: [override.routingId] });
+        if (search.isErr()) {
+            return Err(search.error);
+        }
+        const existing = search.value.get(override.routingId);
+
+        const image = override.image ?? existing?.image;
+        const isDefault =
+            !image &&
+            defaultConfig.cpuMilli === override.cpuMilli &&
+            defaultConfig.memoryMb === override.memoryMb &&
+            defaultConfig.storageMb === override.storageMb;
+
+        if (isDefault) {
+            return nodeConfigOverrides.remove(this.dbClient.db, override.routingId);
+        }
+
+        return nodeConfigOverrides.upsert(this.dbClient.db, override);
+    }
```

Comment:
should this be in a transaction? fleet could modify or access this in between calls no?

Reply:
indeed. concurrent calls to this functions could lead to race condition. Let me add a `FOR UPDATE` and transaction